src/picoprobe_config.h

```diff
@@ -30,5 +30,7 @@
 // UART config
 #define PICOPROBE_UART_TX 4
 #define PICOPROBE_UART_RX 5
```

Comment:
This refers to the base code too, but what is the benefit in general of not using

PICO_DEFAULT_UART_BAUD_RATE (_TX_PIN, _RX_PIN etc) as the default values for PICOPROBE_ values unless they are defined by the user 

i;.e 

```
#ifndef PICOPROBE_UART_TX
#define PICOPROBE_UART_TX PICO_DEFAULT_UART_TX_PIN
#endif
```

which i'd prefer

Reply:
I guess the answer to that depends on why @liamfraser opted to use UART pins 4 and 5 in the first place, instead of just sticking to the default 0 and 1 ? :man_shrugging:

Reply:
Also, if picoprobe_config.h switches to using
```
#ifndef PICOPROBE_UART_TX
#define PICOPROBE_UART_TX PICO_DEFAULT_UART_TX_PIN
#endif
```
where does the definition of a non-default PICOPROBE_UART_TX move to? CMakeLists.txt? Only allow it to be specified on the `cmake` command line? (I'm not being facetious, I'm genuinely curious)

Reply:
yes it would from CMakeLists.txt (which could (should) choose to pick it up from the command line or environment)

Reply:
The reason it is done like this is so you can use the default uart as a debug uart and uart1 as the uart that binds to usb

Reply:
I've never written any CMake before. But poking around in some of the other cmake files, would I want to do something like this?
```cmake
if (DEFINED ENV{PICOPROBE_UART_TX} AND (NOT PICOPROBE_UART_TX))
    set(PICOPROBE_UART_TX $ENV{PICOPROBE_UART_TX})
    message("Using PICOPROBE_UART_TX from environment (${PICOPROBE_UART_TX})")
else ()
    set(PICOPROBE_UART_TX 4)
endif ()
```
:shrug: